src/Tools/PrepareTests/MinimizeUtil.cs

```diff
@@ -40,59 +60,45 @@ void initialWalk()
+                if (currentOutputDirectory != lastOutputDirectory)
                 {
-                    var currentDirName = Path.GetDirectoryName(sourceFilePath)!;
-                    var currentRelativeDirectory = Path.GetRelativePath(sourceDirectory, currentDirName);
-                    var currentOutputDirectory = Path.Combine(destinationDirectory, currentRelativeDirectory);
                     Directory.CreateDirectory(currentOutputDirectory);
-                    var fileName = Path.GetFileName(sourceFilePath);
-
-                    if (fileName.EndsWith(".dll") && TryGetMvid(sourceFilePath, out var mvid))
-                    {
-                        if (!idToFilePathMap.TryGetValue(mvid, out var list))
-                        {
-                            list = new List<FilePathInfo>();
-                            idToFilePathMap[mvid] = list;
-                        }
-
-                        var filePathInfo = new FilePathInfo(
-                            RelativeDirectory: currentRelativeDirectory,
-                            Directory: currentDirName,
-                            RelativePath: Path.Combine(currentRelativeDirectory, fileName),
-                            FullPath: sourceFilePath);
-                        list.Add(filePathInfo);
-                    }
-                    else
-                    {
-                        var destFilePath = Path.Combine(currentOutputDirectory, fileName);
-                        CreateHardLink(destFilePath, sourceFilePath);
-                    }
+                    lastOutputDirectory = currentOutputDirectory;
                 }
-            }
-
-            // https://github.com/dotnet/roslyn/issues/49486
-            // we should avoid copying the files under Resources.
-            var individualFiles = new[]
-            {
-                "global.json",
-                "NuGet.config",
-                "src/Workspaces/MSBuildTest/Resources/.editorconfig",
-                "src/Workspaces/MSBuildTest/Resources/global.json",
-                "src/Workspaces/MSBuildTest/Resources/Directory.Build.props",
-                "src/Workspaces/MSBuildTest/Resources/Directory.Build.targets",
-                "src/Workspaces/MSBuildTest/Resources/Directory.Build.rsp",
-                "src/Workspaces/MSBuildTest/Resources/NuGet.Config",
-            };
+                var fileName = Path.GetFileName(sourceFilePath);
 
-            foreach (var individualFile in individualFiles)
-            {
-                var outputPath = Path.Combine(destinationDirectory, individualFile);
-                var outputDirectory = Path.GetDirectoryName(outputPath)!;
-                Directory.CreateDirectory(outputDirectory);
-                CreateHardLink(outputPath, Path.Combine(sourceDirectory, individualFile));
+                if (fileName.EndsWith(".dll") && TryGetMvid(sourceFilePath, out var mvid))
```

Comment:
```suggestion
                if (fileName.EndsWith(".dll", StringComparison.Ordinal) && TryGetMvid(sourceFilePath, out var mvid))
```

The default for `StartsWith` and `EndsWith` is not ordinal.